packages/truffle-interface-adapter/lib/web3-shim.ts

```diff
@@ -56,33 +65,13 @@ export class Web3Shim extends Web3 {
     this.initInterface();
   }
 
-  initInterface() {
-    switch (this.networkType) {
-      case "quorum": {
-        this.initQuorum();
-        break;
-      }
-      case "ethereum":
-      default: {
-        this.initEthereum();
-        break;
-      }
-    }
-  }
-
-  initEthereum() {
-    // truffle has started expecting gas used/limit to be
-    // hex strings to support bignumbers for other ledgers
-    ethereumOverloads.getBlock(this);
-    ethereumOverloads.getTransaction(this);
-    ethereumOverloads.getTransactionReceipt(this);
-  }
+  async initInterface() {
+    const networkTypes: NetworkTypesConfig = new Map(Object.entries({
+      "ethereum": EthereumDefinition,
+      "quorum": QuorumDefinition,
+      "fabric-evm": FabricEvmDefinition
+    }));
```

Comment:
can we move this block to _immediately_ follow the imports, as a module-scoped constant? (Same reasoning as the other comment: make the code surface area as small as possible when adding a new network type)